src/2.0/types.ts

```diff
@@ -9,12 +10,16 @@ export interface Proof {
   verificationMethod: string;
   signature: string;
 }
-export interface Signature {
-  type: "SHA3MerkleProof";
-  targetHash: string;
-  proof: string[];
-  merkleRoot: string;
-}
+
+const hashRegex = /^[0-9A-Fa-f]{64}$/;
+const Hash = String.withConstraint(value => hashRegex.test(value));
```

Comment:
it's really a 32 byte unprefixed hexstring and each byte is represented with 2 hexadecimal chars hence 64 length
(ethers has a type/util for 0x prefixed https://docs.ethers.io/v5/api/utils/bytes/#HexString)

my issue with this is mostly the ambiguous name of 'Hash'

Reply:
also in withConstraint() you can return a string which will be thrown as the error when validation fails
e.g

String.withConstraint((e) => e.length === 64 || 'e is not the expected length of 64')  

more info about this here https://github.com/pelotom/runtypes#constraint-checking

Reply:
- renamed to OpenAttestationHexString
- added error message